cmd/spire-agent/cli/run/run_posix_test.go

```diff
@@ -5,15 +5,121 @@ package run
 
 import (
 	"bytes"
+	"fmt"
+	"io"
 	"os"
+	"syscall"
 	"testing"
 
 	"github.com/hashicorp/hcl/hcl/printer"
 	"github.com/spiffe/spire/pkg/agent"
+	commoncli "github.com/spiffe/spire/pkg/common/cli"
+	"github.com/spiffe/spire/pkg/common/log"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func TestCommand_Run(t *testing.T) {
+	testTempDir := t.TempDir()
+	testDataDir := fmt.Sprintf("%s/data", testTempDir)
+	testAgentSocketDir := fmt.Sprintf("%s/spire-agent", testTempDir)
+
+	type fields struct {
+		logOptions         []log.Option
+		env                *commoncli.Env
+		allowUnknownConfig bool
+	}
+	type args struct {
+		args []string
+	}
+	type want struct {
+		code               int
+		dataDirCreated     bool
+		agentUdsDirCreated bool
+	}
+	tests := []struct {
+		name     string
+		osTarget string
+		fields   fields
+		args     args
+		want     want
+	}{
+		{
+			name:     "don't create any dir when error loading config",
+			osTarget: "linux",
+			args: args{
+				args: []string{},
+			},
+			fields: fields{
+				logOptions: []log.Option{},
+				env: &commoncli.Env{
+					Stderr: io.Discard,
+				},
+				allowUnknownConfig: false,
+			},
+			want: want{
+				code:               1,
+				agentUdsDirCreated: false,
+				dataDirCreated:     false,
+			},
+		},
+		{
+			name:     "creates spire-agent uds and data dirs",
```

Comment:
wasn't able to explore more scenarios loading config successfully due to fflag singleton (after it's first instantiation, no flags can be loaded anymore, ref pkg/common/fflag/fflag.go:67) Any suggestions?

Reply:
I think that running `os.Exit(0)` at the end of each test run should do the trick. By default, golang will panic unless `test.paniconexit0` is set to false. I'm curious to know if @azdagron has a suggestion here.

Reply:
We discussed this in the contributors sync, and the consensus was that enhancing the fflag package to be able to externally set `singleton.loaded` to false is probably the best solution for this.

Reply:
I will proceed by extending fflag package with this feature 👍